cilium-cli/connectivity/check/deployment.go

```diff
@@ -1179,6 +1135,255 @@ func (ct *ConnectivityTest) deploy(ctx context.Context) error {
+		}
+	}
+
+	// Make sure that the server deployment is ready to spread client connections
+	err = WaitForDeployment(ctx, ct, ct.clients.src, ct.params.TestNamespace, deployName)
+	if err != nil {
+		ct.Failf("%s deployment is not ready: %s", deployName, err)
+	}
+
+	for _, client := range ct.Clients() {
+		_, err = client.GetService(ctx, ct.params.TestNamespace, svcName, metav1.GetOptions{})
+		if err != nil {
+			ct.Logf("✨ [%s] Deploying %s service...", client.ClusterName(), svcName)
+			svc := newService(svcName, map[string]string{"app": appLabel}, nil, "http", 8000, ct.Params().ServiceType)
+			svc.ObjectMeta.Annotations = map[string]string{"service.cilium.io/global": "true"}
+			_, err = client.CreateService(ctx, ct.params.TestNamespace, svc, metav1.CreateOptions{})
+			if err != nil {
+				return fmt.Errorf("unable to create service %s: %w", svcName, err)
+			}
+		}
+
+		if enabled, _ := ct.Features.MatchRequirements(features.RequireEnabled(features.CNP)); enabled {
+			ipsec, _ := ct.Features.MatchRequirements(features.RequireMode(features.EncryptionPod, "ipsec"))
+			if ipsec && versioncheck.MustCompile(">=1.14.0 <1.16.0")(ct.CiliumVersion) {
+				// https://github.com/cilium/cilium/issues/36681
+				continue
+			}
+			for _, client := range ct.Clients() {
+				cnp := cnpFunc(ct.params.TestNamespace)
+				ct.Logf("✨ [%s] Deploying %s CiliumNetworkPolicy...", client.ClusterName(), cnp.Name)
+				_, err = client.ApplyGeneric(ctx, cnp)
+				if err != nil {
+					return fmt.Errorf("unable to create CiliumNetworkPolicy %s: %w", cnp.Name, err)
+				}
+			}
+		}
+	}
+
+	return err
+}
+
+func (ct *ConnectivityTest) createTestConnDisruptClientDeployment(ctx context.Context, deployName, kind, appLabel, address string, replicas int, isExternal bool) error {
+	_, err := ct.clients.dst.GetDeployment(ctx, ct.params.TestNamespace, deployName, metav1.GetOptions{})
+	if err != nil {
+		ct.Logf("✨ [%s] Deploying %s deployment...", ct.clients.dst.ClusterName(), deployName)
+		readinessProbe := &corev1.Probe{
+			ProbeHandler: corev1.ProbeHandler{
+				Exec: &corev1.ExecAction{
+					Command: []string{"cat", "/tmp/client-ready"},
+				},
+			},
+			PeriodSeconds:       int32(3),
+			InitialDelaySeconds: int32(1),
+			FailureThreshold:    int32(20),
+		}
+
+		param := deploymentParameters{
+			Name:     deployName,
+			Kind:     kind,
+			Image:    ct.params.TestConnDisruptImage,
+			Replicas: replicas,
+			Labels:   map[string]string{"app": appLabel},
+			Command: []string{
+				"tcd-client",
+				"--dispatch-interval", ct.params.ConnDisruptDispatchInterval.String(),
+				address,
+			},
+			ReadinessProbe: readinessProbe,
+			Resources: corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{corev1.ResourceCPU: *resource.NewMilliQuantity(100, resource.DecimalSI)},
+			},
+		}
+		if isExternal {
+			param.NodeSelector = map[string]string{"cilium.io/no-schedule": "true"}
```

Comment:
We have a constant for this label: https://github.com/cilium/cilium/blob/main/cilium-cli/defaults/defaults.go#L109

Reply:
Fixed